### ISSUE TITLE
Caching

### ISSUE THREAD

**Issue by till** [galeone/fitbit#9]
I didn't want to create a data layer, hence I am wondering what you think about caching the GET requests? Otherwise, the rate limit happens so quick. ;D I think all that's potentially needed is exposing the http client to register a transport.

**Comment by galeone** [comment]
Are you using the personal access token, right? Otherwise, I find it quite impossible not to implement a persistence layer, since for the OAuth2 flow I require an implementation of `fitbit.Storage` :smile: 

I'm not sure if caching the GET requests can be something totally correct to put inside the client - maybe this should be in the client user?

I'm thinking about the non-idempotent requests, that if cached prevent the client from working correctly. e.g the request for today's activity log is one of these requests. The first time I ask the log, I get a response, the second time I have a different response with the new data gathered in the meantime. The GET URL and the parameters are identical :thinking:

**Comment by till** [comment]
Yeah, I mean, for my use-case _stale_ data would be fine (like updated every 30 minutes or whatever). I wanted to play with it but it would require setting the transport on the client and that's currently not possible, I think. But of course that's up to you if you want to allow it.
